apps/src/aichat/redux/thunks/clearChatMessagesByUser.ts

```diff
@@ -0,0 +1,23 @@
+import {EVENTS} from '@cdo/apps/metrics/AnalyticsConstants';
+import {AppDispatch} from '@cdo/apps/util/reduxHooks';
+
+import {clearChatMessages as clearChatMessagesReducer} from '../slice';
+
+import {addChatEvent} from './addChatEvent';
+import {sendAnalytics} from './sendAnalytics';
+
+// User initiated chat clear: perform state reset, add clear chat event, send analytics.
+export const clearChatMessagesByUser = () => (dispatch: AppDispatch) => {
+  dispatch(clearChatMessagesReducer());
+  dispatch(
+    addChatEvent({
+      timestamp: Date.now(),
+      descriptionKey: 'CLEAR_CHAT',
+    })
+  );
+  dispatch(
+    sendAnalytics(EVENTS.CHAT_ACTION, {
+      action: 'Clear chat history',
```

Comment:
For later, but I imagine we'd eventually want to pass through metadata about the `clientType` to distinguish between AI Tutor, AI Chat, right? Ed and Cassi probably know more about the timeline though 🙂